### PR TITLE
Refactor plot window

### LINKED ISSUE #25: Clicking Display Does bring up a plot after changing the current window
Bring up a slice
 keep it,
 bring up a second
Make the first one current.
Trying to bring up a third slice will fail

### PR DESCRIPTION
**Pull request by EltayebAhmed**
Refactor Plot window to base plot window, 

**To test:**
Load workspace2d in mslice and verify that the plot window works by selecting the workspace and clicking display

Fixes #25

### CODE REVIEW

**Comment by EltayebAhmed** [comment]
Issue #25 was the result of make currenting enabling overplotting,
This issue in no longer present after migrating it to imshow as imshow clears the axes regardless of its state. But the last commit in this branch resolves the issue from the root

File: plotting/plot_window/base_plot_window.py [hunk 1]
```diff
@@ -17,11 +17,9 @@ def __init__(self,number,manager):
...
 
     def set_as_kept(self):
-        self.canvas.axes.hold(False)
         self._display_status("kept")
 
     def set_as_current(self):
-        self.canvas.axes.hold(True)
         self._display_status("current")
 
     def display_status(self,status):
```
Comment (EltayebAhmed) [inline plotting/plot_window/base_plot_window.py]: Issue #25 was the result of make currenting enabling overplotting,
This issue in no longer present after migrating it to imshow as imshow clears the axes regardless of its state. But the last commit in this branch resolves the issue from the root

File: models/projection/powder/mantid_projection_calculator.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from mantid.simpleapi import ConvertToMD
 from models.projection.powder.projection_calculator import ProjectionCalculator
+from models.workspacemanager.mantid_workspace_provider import MantidWorkspaceProvider
```
Comment (martyngigg) [inline models/projection/powder/mantid_projection_calculator.py]: Is this import needed? It doesn't seem to be used anywhere.

**Comment by martyngigg** [comment]
@EltayebAhmed The travis build has failed and there are merge conflicts after the cut tab has been merged to `master`.

**Comment by EltayebAhmed** [comment]
Issues resolved

**Comment by martyngigg** [comment]
Thanks.

A good set of refactorings to pull out common functionality into the base class that knows about the keep/make current status and future scripting. Happy to merge for further testing by scientists.

**Pull request merged**